codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/UnionGenerator.java

```diff
@@ -86,27 +82,23 @@ public void render() {
 
             for (MemberShape memberShape : shape.members()) {
                 writer
-                    .openBlock("class $L < $T",
-                            symbolProvider.toMemberName(memberShape), symbol)
-                    .write("def to_h: () -> { $L: Hash[Symbol,$T] }",
-                            RubyFormatter.toSnakeCase(symbolProvider.toMemberName(memberShape)),
-                            symbol)
+                    .openBlock("class $L < $T", symbolProvider.toMemberName(memberShape), symbol)
+                    .write("def to_h: () -> { $L: Hash[Symbol, $T] }",
```

Comment:
What does this type signature mean? I thought the to_h should potentally be Hash[Symbol, untyped]?

Reply:
This is what it was before. It is the union parent class. It ends up being something like: 

```
    class Union < Hearth::Union
      class String < Union
        def to_h: () -> { string: Hash[Symbol, Union] }
        def to_s: () -> String
      end

      class Struct < Union
        def to_h: () -> { struct: Hash[Symbol, Union] }
        def to_s: () -> String
      end

      class Unknown < Union
        def to_h: () -> { unknown: Hash[Symbol, Union] }
        def to_s: () -> String
      end
    end
```

Reply:
I think actually it needs to be:

```
def to_h: () -> { string: String }
def to_h: () -> { struct: Hash[Symbol, untyped] }
```